src/core/src/graph_util.cpp

```diff
@@ -749,51 +749,65 @@ bool ngraph::check_for_cycles(const ngraph::Function* func, ngraph::NodeVector&
     return false;
 }
 
+bool has_result_consumers(const Output<Node>& port) {
+    const auto& consumers = port.get_target_inputs();
+    return std::any_of(consumers.cbegin(), consumers.cend(), [](const Input<Node>& consumer) {
+        return ov::is_type<op::v0::Result>(consumer.get_node());
+    });
+};
+
+bool ov::replace_outputs_update_names(OutputVector outputs, const OutputVector& replacements) {
```

Comment:
Changes in this file will be moved to the separate PR. 
This is not a part of the transformation.